src/ucp/wireup/wireup_ep.c

```diff
@@ -168,14 +171,21 @@ static ucs_status_t ucp_wireup_ep_pending_add(uct_ep_h uct_ep,
                                               unsigned flags)
 {
     ucp_wireup_ep_t *wireup_ep = ucp_wireup_ep(uct_ep);
-    ucp_ep_h ucp_ep = wireup_ep->super.ucp_ep;
-    ucp_worker_h worker = ucp_ep->worker;
+    ucp_ep_h ucp_ep            = wireup_ep->super.ucp_ep;
+    ucp_worker_h worker        = ucp_ep->worker;
     ucp_request_t *proxy_req;
+    ucp_request_t UCS_V_UNUSED *msg_req;
     uct_ep_h wireup_msg_ep;
     ucs_status_t status;
 
     UCS_ASYNC_BLOCK(&worker->async);
     if (req->func == ucp_wireup_msg_progress) {
+        msg_req = ucs_container_of(req, ucp_request_t, send.uct);
+        ucs_assertv(msg_req->send.wireup.type != UCP_WIREUP_MSG_ACK,
```

Comment:
also check `EP_CHECK/EP_REMOVED`? Or maybe better check what msg types can be here?

Reply:
Actually, REMOVED/EP_CHECK can be here during connection establishment in case of RC. Only ACK shouldn’t be added to pending queue of WIREUP EP in any case.